runtime/runtime/src/bandwidth_scheduler/mod.rs

```diff
@@ -38,55 +48,89 @@ pub fn run_bandwidth_scheduler(
         Some(prev_state) => prev_state,
         None => {
             tracing::debug!(target: "runtime", "Bandwidth scheduler state not found - initializing");
-            BandwidthSchedulerState { mock_data: [0; 32] }
+            BandwidthSchedulerState {
+                link_allowances: Vec::new(),
+                sanity_check_hash: CryptoHash::default(),
+            }
         }
     };
 
-    let scheduler_state_hash: CryptoHash = hash(&borsh::to_vec(&scheduler_state).unwrap());
+    // Prepare the status info for each shard based on congestion info.
+    let mut shards_status: BTreeMap<ShardId, ShardStatus> = BTreeMap::new();
+    for (shard_id, extended_congestion_info) in apply_state.congestion_info.iter() {
+        let last_chunk_missing = extended_congestion_info.missed_chunks_count > 0;
+        let is_allowed_sender_shard =
+            *shard_id == ShardId::from(extended_congestion_info.congestion_info.allowed_shard());
+
+        let congestion_level = CongestionControl::new(
+            apply_state.config.congestion_control_config,
+            extended_congestion_info.congestion_info,
+            extended_congestion_info.missed_chunks_count,
+        )
+        .congestion_level();
+        let is_fully_congested = CongestionControl::is_fully_congested(congestion_level);
+
+        shards_status.insert(
+            *shard_id,
+            ShardStatus { last_chunk_missing, is_allowed_sender_shard, is_fully_congested },
+        );
+    }
 
+    // Prepare lists of sender and receiver shards.
+    // TODO(bandwidth_scheduler) - find a better way to get the sender and receiver shards.
+    // Maybe pass shard layout in ApplyState? That might also be needed for resharding.
+    // Taking all shards from the congestion info will be wrong for only one height during
+    // resharding, so it might be good enough for now, but it's not ideal.
     let mut all_shards = apply_state.congestion_info.all_shards();
     all_shards.sort();
     if all_shards.is_empty() {
-        all_shards = vec![ShardId::new(0)];
+        // In some tests there's no congestion info and the list of shards ends up empty.
+        // Pretend that there's only one default shard.
+        all_shards = vec![ShardUId::single_shard().shard_id()];
```

Comment:
The single_shard should only be used for tests. This shard uid is not valid for production shard layout and it may change in the future.

Reply:
Hopefully this will be removed after switching to `ShardLayout` for the list of shards, for now it's needed to make some tests pass.

Reply:
It has been removed with the move to `ShardLayout`, marking as resolved